collector/src/main/java/com/bitmovin/analytics/BitmovinAdAnalytics.kt

```diff
@@ -4,32 +4,39 @@ import com.bitmovin.analytics.ads.*
 import com.bitmovin.analytics.data.AdSample
 import com.bitmovin.analytics.data.AdEventData
 import com.bitmovin.analytics.utils.Util
-import java.util.Collections.min
 
 class BitmovinAdAnalytics(var analytics: BitmovinAnalytics) {
     private var activeAdBreak: AdBreak? = null
     private var activeAdSample: AdSample? = null
     private var adPodPosition: Int = 0
     private var adStartupTimestamp: Long? = null
-    private var currentTime: Long? = null
+    private var _currentTime: Long? = null
     private var beginPlayingTimestamp: Long? = null
     private var isPlaying: Boolean = false
     private val adManifestDownloadTimes: HashMap<String, Long> = hashMapOf()
 
+    private val currentTime: Long?
+        get() = if(this.isPlaying) {
+            if(this._currentTime == null || this.beginPlayingTimestamp == null) {
+                null
+            } else {
+                this._currentTime!! + Util.getTimeStamp() - this.beginPlayingTimestamp!!
+            }
+
+        } else {
+            _currentTime
+        }
+
     fun onAdStarted(ad: Ad) {
         if (!ad.isLinear) {
             return
         }
 
         this.resetActiveAd()
         this.activeAdSample = AdSample(ad = ad)
+        this.activeAdSample!!.adStartupTime = if (this.adStartupTimestamp != null) Util.getTimeStamp() - this.adStartupTimestamp!! else null
 
-        val activeAdSample = this.activeAdSample ?: return
-
-        val adStartupTime = this.adStartupTimestamp
-        activeAdSample.adStartupTime = if (adStartupTime != null) Util.getTimeStamp() - adStartupTime else null
-
-        this.startAd(activeAdSample)
+        this.startAd(activeAdSample!!)
```

Comment:
@Tigraine @rolandgriesser would it be more secure, regarding an NullPointerException, if we assign the value of the `this.acticsAdSample` to a new variable, exit if the value is null and proceed with the local variable

like `val newAdSample = this.activeAdSample ?: return`